### PR TITLE
php8 fix

### LINKED ISSUE #1942: Deprecated Functionality: number_format(): Passing null to parameter #1 ($num) of type float is deprecated
Hi Adyen,

We are seeing the following message appearing in our order history for MobilePay orders

Deprecated Functionality: number_format(): Passing null to parameter #1 ($num) of type float is deprecated in ... vendor/adyen/module-payment/Helper/Data.php on line 372

We are using version 8.13 of the Adyen module and PHP 8.1

Thanks,
Ryan

### PR DESCRIPTION
**Pull request by skeutmeier**
**Description**

fix for PHP 8. If Products without a price are in cart, it will result in a NULL parameter on number_format, which is not allowed anymore in PHP 8

**Tested scenarios**

added bundle products to cart and be able to checkout without php error

Fixes  
fixes #1942

### CODE REVIEW

File: Helper/Data.php [hunk 1]
```diff
@@ -372,6 +372,7 @@ public function decimalNumbers($currency)
      */
     public function formatAmount($amount, $currency)
     {
+        if ($amount == null) {$amount = 0};	// PHP 8 does not accept first param to be NULL
```
Comment (peterojo) [inline Helper/Data.php:375]:
```suggestion
        if ($amount === null) {
            // PHP 8 does not accept first param to be NULL
            $amount = 0;
        }
```

**Review by peterojo — APPROVED** [review]

**Comment by skeutmeier** [comment]
Is there something i can do to help push this through?

**Comment by Morerice** [comment]
Hi @skeutmeier,

Apologies for the delayed response and thanks for opening this PR. We will need to investigate this longer to ensure that this does not affect other functionality. Because there are issues related to this already (#1958) merging this PR may result in cases where this function returns 0, in cases where there is an actual amount that should be getting passed. However I understand that this will resolve the specific issue you're mentioning.

We will be creating an internal ticket to investigate a solution to your issue, ideally by mitigating it before the `number_format` functionality is called.

Regards,
Jean
Adyen

**Comment by f123248** [comment]
Hi @Morerice ,

May I kindly ask how is the investigate going on

**Comment by skeutmeier** [comment]
> Hi @skeutmeier,
> 
> Apologies for the delayed response and thanks for opening this PR. We will need to investigate this longer to ensure that this does not affect other functionality. Because there are issues related to this already (#1958) merging this PR may result in cases where this function returns 0, in cases where there is an actual amount that should be getting passed. However I understand that this will resolve the specific issue you're mentioning.
> 
> We will be creating an internal ticket to investigate a solution to your issue, ideally by mitigating it before the `number_format` functionality is called.
> 
> Regards, Jean Adyen

any updates on this?

**Review by peterojo — APPROVED** [review]

**Pull request merged**